homeassistant/components/webdav/camera.py

```diff
@@ -0,0 +1,235 @@
+
+    @property
+    def supported_features(self):
+        """Return supported features."""
+        return [SUPPORT_ON_OFF]
+
+    def update(self):
+        """Fetch the current contents of the file share."""
+        try:
+            self._files = [
+                filename
+                for filename in self._client.list(self._directory)
+                if not filename.endswith("/")
+            ]
+            self._files.sort()
+            if not self._available:
+                self._available = True
+                _LOGGER.info("Reconnected to WebDAV camera %s", self.name)
+        except WebDavException as exception:
+            self._files = []
+            if self._available:
+                self._available = False
+                _LOGGER.warning(
+                    "Could not open WebDAV camera %s. Message: %s", self.name, exception
+                )
+
+    @property
+    def available(self):
+        """Return True if the file share is available and contains images."""
+        return self._available and self._has_images
+
+    @property
+    def name(self):
+        """Return the name of this entity."""
+        return self._name
+
+    async def async_camera_image(self):
+        """Fetch the current image from the share."""
+        with await self._image_lock:
+            if self._image is not None:
+                return self._image
+            file_name = self._image_url
+
+            try:
+                async with self._session.get(file_name) as resp:
+                    resp.raise_for_status()
+                    # Cache the image; that way streams don't fetch it too much
+                    self._image = await resp.read()
+            except ClientError as error:
+                _LOGGER.warning("Failed to download %s: %s", file_name, error)
+                self._available = False
+
+            return self._image
+
+    @property
+    def is_on(self):
+        """Return True if the camera is playing through files in the share."""
+        return self._stop_advancing is not None
+
+    def turn_on(self):
+        """Start playing through files in the share."""
+        if not self._stop_advancing:
+            self._stop_advancing = async_track_time_interval(
+                self.hass, self._advance, self._image_interval
+            )
+
+    def turn_off(self):
+        """Stop playing through files in the share."""
+        if self._stop_advancing:
+            self._stop_advancing()
+            self._stop_advancing = None
+
+    @property
+    def _image_filename(self):
+        return self._files[self._image_number % len(self._files)]
+
+    @property
+    def _image_url(self):
+        return self._client.get_url(self._directory + self._image_filename)
+
+    async def _advance(self, _=None):
+        images_checked = 0
+        while images_checked < len(self._files):
+            self._image_number += 1
+            images_checked += 1
+            content_type = self._client.get_property(
```

Comment:
This makes I/O so needs to be scheduled on the executor thread pool. Use `await self.hass.async_add_executor_job(...)`.